lib/broadway_kafka/producer.ex

```diff
@@ -323,13 +324,10 @@ defmodule BroadwayKafka.Producer do
          not is_draining_after_revoke?(state.draining_after_revoke_flag) and
          offset != nil do
       messages = fetch_messages_from_kafka(state, key, offset)
-      to_send = min(demand, max_demand)
-      {new_acks, not_sent, messages, pending} = split_demand(messages, acks, key, to_send)
```

Comment:
I've decided to not send any events after fetching events and let `maybe_schedule_poll` take care of dispatching events